src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala

```diff
@@ -0,0 +1,33 @@
+package com.sergigp.quasar.event
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+
+import org.slf4j.Logger
+
+class AsyncEventBus(logger: Logger)(implicit ec: ExecutionContext) extends EventBus[Future] {
+
+  private var handlers = Map.empty[Class[_], Event => Future[Any]]
+
+  override def publish[E <: Event](command: E): Future[E#EventReturnType] =
+    handlers
+      .get(command.getClass) match {
+      case Some(handler) => handler(command).map(_.asInstanceOf[E#EventReturnType])
+      case None          => Future.failed(EventHandlerNotFound(command.getClass.getSimpleName))
+    }
+
+  override def subscribe[E <: Event: ClassTag](handler: E => Future[E#EventReturnType]): Unit = {
```

Comment:
Instead of a `subscribe` method we could pass it via constructor 🤔

Reply:
circular dependency if you want to chain events